hail/src/main/scala/is/hail/expr/ir/agg/CollectAsSetAggregator.scala

```diff
@@ -162,7 +162,7 @@ class CollectAsSetAggregator(t: PType) extends StagedAggregator {
     other.foreach { (km, kv) => state.insert(km, kv) }
 
   def result(state: State, srvb: StagedRegionValueBuilder, dummy: Boolean): Code[Unit] =
-    srvb.addArray(resultType.fundamentalType.asPArray, sab =>
+    srvb.addArray(resultType.arrayFundamentalType.asPArray, sab =>
```

Comment:
do we need asParray for this and the below? asParray is really a shallow copy of a PArray (calls `PArray(this.elementType, this.required)`, so I think it could be more clearly handled by ".copy()", or if we don't nee d the copy, by just `resultType.arrayFundamentalType`

Reply:
no, I don't think so.

Reply:
removed